### ISSUE TITLE
t1-freesurfer is not grabbing correctly input files

### ISSUE THREAD

**Issue by arnaudmarcoux** [aramis-lab/clinica#34]
This is a recurring problem with BIDS and CAPS layout :
In t1-freesurfer : 

```python 
        # Remove 'sub-' prefix from participant IDs
        participant_labels = '|'.join(sub[4:] for sub in self.subjects)
        # Remove 'ses-' prefix from session IDs
        session_labels = '|'.join(ses[4:] for ses in self.sessions)

        error_message = ""
        # Inputs from anat/ folder
        # ========================
        # T1w file:
        t1w_files = self.bids_layout.get(type='T1w', return_type='file', extensions=['.nii|.nii.gz'],
                                         subject=participant_labels, session=session_labels)
        error_message += check_input_bids_files(t1w_files, "T1W_NII", self.bids_directory, self.subjects, self.sessions)
```

Bids layout acts as a succession of filters, so in the case where tsv file is : 
sub-01 ses-M00
sub-02 ses-M00
sub-02 ses-M06

If sub-01 has a ses-M06 session, it will also be catched by the bids layout

**Comment by alexandreroutier** [comment]
Hi!

Thanks for the feedback.

Normally, this should not occur for CAPS Layout. Indeed, in `check_input_caps_files` function, the following line:
```python
list_caps_files = reorder_bids_or_caps_files(subject_ids, list_caps_files)
```
aims to solve the issue you descibed. However, this is indeed missing in `check_input_bids_files` function. This will be solved in PR #36.

Alexandre

**Comment by arnaudmarcoux** [comment]
I did not catch the line at the end of check_inputs_caps_file,my bad ! Still I think it is strange that this function does not return the list it is modifying, isn't  it ? I know it can be done, but it is not clear that this function is modifying the file list.

I think a `error_message, t1w_files = check_input_bids_files(t1w_files, "T1W_NII"...)` would be clearer. What do you think ?

BTW, I am also currently working on a solution to remove grabbit library in https://github.com/aramis-lab/clinica/pull/35, so that we won't have to face this issue anymore.

**Comment by alexandreroutier** [comment]
> I did not catch the line at the end of check_inputs_caps_file,my bad ! Still I think it is strange that this function does not return the list it is modifying, isn't it ? I know it can be done, but it is not clear that this function is modifying the file list.

This is an old habit from C/C++ language that I should remove, I agree with you.

> I think a `error_message, t1w_files = check_input_bids_files(t1w_files, "T1W_NII"...)` would be clearer. What do you think ?
> 
> BTW, I am also currently working on a solution to remove grabbit library in #35, so that we won't have to face this issue anymore. Of course we'll have time to discuss this in near future.

I agree! :) The only issue is that I can not append the error message when checking BIDS/CAPS files. Since you are currently working on that, I think we add this point to the Clinica meeting Thursday. Keep in mind that when reading several types of BIDS/CAPS files, we need to be able to:
- handle cases where several images are found (not handle by Clinica for BIDS, abnormal situation for CAPS) with an exception
- handle cases where images are missing with an exception. (For CAPS, we also need to explain which pipeline the user did not run)
- handle `group_id` and  `long_id`. It can be new functions having the same code structure (they extend somehow the notion of participants and sessions)

The first two points are already covered in `check_{bids|caps}` function in `utils/io.py`.

**Comment by alexandreroutier** [comment]
Currently solved in #36. Will be definitely and properly solved in #35.
